### ISSUE TITLE
react-sortable-hoc broken in ConcurrentMode

### ISSUE THREAD

**Issue by Kovensky** [clauderic/react-sortable-hoc#460]
I haven't really made a reproduction case yet, but just enabling `ConcurrentMode` in my tree that includes `react-sortable-hoc` is sufficient to break it as it is used in my application.

Related to https://github.com/clauderic/react-sortable-hoc/issues/444.

When `ConcurrentMode` is enabled, the helpers that show up when dragging starts never disappear, and the SortableList also does not correctly update when a reorder does happen.

**Comment by clauderic** [comment]
Hey @Kovensky,

That's no good, I haven't tried using `react-sortable-hoc` with `ConcurrentMode` yet, but I will investigate. 

You mentioned #444, what makes you think the issues you noticed are related to that issue?

**Comment by Kovensky** [comment]
`SortableElement` does side effects in `componentWillReceiveProps`, which is exactly the kind of thing that is expected to break in `ConcurrentMode` and why it is a `StrictMode` warning. I believe what is happening is a race condition between `SortableElement` removing itself from the `Manager` and `SortableContainer`'s `handleSortEnd` logic being invoked; as `SortableElement#componentWillReceiveProps` might be invoked multiple times without committing to the DOM as part of a single yieldy render.

**Comment by clauderic** [comment]
Right. I have a PR open that updates the behaviour of the `disabled` prop, and also changes from using `componentWillReceiveProps` to `componentDidUpdate` in `SortableElement` to manage the side-effects. I haven't tested it with ConcurrentMode yet, but I'm expecting it should work properly.

**Comment by Kovensky** [comment]
I haven't tested it but it does look more correct 👍 (fixed link: https://github.com/clauderic/react-sortable-hoc/pull/472)

As for the remaining warnings:

- String ref can probably easily be fixed by using a function ref (or `createRef`, but if you do that you might as well do the legacy context migration at the same time).
- Legacy context is more of a performance / correctness issue but I don't think it'll break in `ConcurrentMode`, as long as context updates only happen in commit lifecycles (`componentDidMount` / `componentDidUpdate` / `componentWillUnmount`). AFAICT context updates don't really happen at all (`createContext`-wise); the `manager` is created once and is only internally mutated.
- `findDOMNode` is deprecated because it only returns the first DOM node for components that produce more than one, but it otherwise should be thread-safe to call in `componentDidMount` / `componentDidUpdate`. It's still bad to call it in other lifecycles, in render, or in event handlers.
